### PR TITLE
[release 3.11] fix OpenStack heat template conditional

### PR DESCRIPTION
**Pull request by tzumainn**
Correct heat template conditional, separating out the provider network bits from the kuryr bits.

### CODE REVIEW

File: roles/openshift_openstack/templates/heat_stack.yaml.j2 [hunk 1]
```diff
@@ -226,6 +226,40 @@ resources:
...
+{% if not openshift_openstack_node_subnet_name %}
+  net:
+    type: OS::Neutron::Net
+    properties:
+      name:
+        str_replace:
+          template: openshift-ansible-cluster_id-net
+          params:
+            cluster_id: {{ openshift_openstack_full_dns_domain }}
+{% endif %}
```
Comment (luis5tb) [inline roles/openshift_openstack/templates/heat_stack.yaml.j2]: perhaps only one if (at line 230) for this, right

File: roles/openshift_openstack/templates/heat_stack.yaml.j2 [hunk 2]
```diff
@@ -226,6 +226,40 @@ resources:
 {% endif %}
 
 {% if not openshift_openstack_provider_network_name %}
+
+{% if not openshift_openstack_node_subnet_name %}
```
Comment (luis5tb) [inline roles/openshift_openstack/templates/heat_stack.yaml.j2]: why it is not enough with checking openshift_openstack_provider_network_name?

**Comment by tzumainn** [comment]
@luis5tb Thanks for the review! I've updated the PR - let me know what you think!

File: roles/openshift_openstack/templates/heat_stack.yaml.j2 (hunk 2 above)
Comment (luis5tb) [reply roles/openshift_openstack/templates/heat_stack.yaml.j2]: got it! I also see that you limited the if openshift_openstack_provider_network_name block to just the net/subnet creation, which seems right. Thanks!

**Pull request merged**